explainer.md

```diff
@@ -395,13 +395,143 @@ function checkMagicWindowSupport() {
+
+```js
+// Loop over every input source and get their pose for the current frame.
+for (let inputSource of xrInputSources) {
+  let inputPose = xrFrame.getInputPose(inputSource, xrFrameOfRef);
+
+  // Check to see if the pose is valid
+  if (inputPose) {
+    // Render a visualization of the input source (see next section).
+    RenderInputSource(inputSource, inputPose);
+
+    // Highlight any objects that the pointing ray intersects with.
+    // Presumes the use of a fictionalized rendering library.
+    let hoveredObject = scene.getObjectIntersectingRay(inputPose.pointerMatrix);
+    if (hoveredObject) {
+      renderer.drawHighlight(hoveredObject);
+    }
+  }
+}
+```
+
+In some cases tracked input sources cannot accurately track their position in space, and provides an estimated position based on the sensor data available to it. This is the case, for example, for the Daydream and GearVR 3DoF controllers, which use an arm model to approximate controller position based on rotation. In these cases the `emulatedPosition` attribute of the `XRInputPose` should be set to `true` to indicate that the translational components of the pose matrices may not be accurate.
+
+### Rendering input sources
+
+Most applications will want to visually represent the input sources somehow. The appropriate type of visualization to be used depends on the value of the `pointerOrigin` attribute:
+
+  * `'head'`: A cursor should be drawn at some distance down the pointer ray, ideally at the depth of the first surface it intersects with, so the user can identify what will be interacted with when a select event is fired. It's not appropriate to draw a controller or ray in this case, since they may obscure the user's vision or be difficult to visually converge on.
+  * `'hand'`: If the `gripMatrix` in not `null` an application-appropriate controller model should be drawn using that matrix as the transform. If appropriate for the experience, the a visualization of the pointer ray and a cursor as described in the `'head'` should also be drawn.
+
+`getInputSources()` will never return input sources with a `pointerOrigin` of `screen`, so they do not need to be accounted for when rendering.
+
+```js
+// Render a visualization of the input source.
+// Presumes the use of a fictionalized rendering library.
+function RenderInputSource(inputSource, inputPose) {
+  // If the input source is a handheld device with a valid gripMatrix, render
+  // a controller mesh using the gripMatrix as a 
+  if (inputSource.pointerOrigin == "hand" && inputPose.gripMatrix) {
+    let controllerMesh = getControllerMesh();
```

Comment:
Per today's call, XRInputSource could expose some kind of modelId or renderModelId property that could be passed to getControllerMesh here, to enable a request to a CDN.  This could be some kind of common string form of VID/PID for motion controllers, or just the string "hand" for hand sources.  If we are concerned about use of this as a general controller identifier, we could perhaps leave the id blank for controllers with no gripMatrix (and thus no 6DoF controller rendering).

Reply:
Per the discussion below, UAs could return a renderModelId of null on headsets where you can see your hands/controller and a renderModelId of "hand" or a VID/PID when you can't see your hands/controller.  That would be a strong hint to the app as to whether it should render the controller.